### ISSUE TITLE
SimpleSelect doesn't have aria-role of listbox

### ISSUE THREAD

**Issue by buckett** [instructure/instructure-ui#1132]
### Question

We are doing some testing of our UIs that are built with Instructure UI and I was just trying to test a `<SimpleSelect>` component using [Testing Library](https://testing-library.com/). When using this library your encouraged to think about the aria role of the component you are trying to interact with. I expected the `<SimpleSelect>` component to be a [listbox](https://w3c.github.io/aria/#listbox) but instead is seems to be a [button](https://w3c.github.io/aria/#button).

Is there a reason there aren't elements on the page that have a role of `listbox`?

Instead I am driving the component more specifically with:
```
        const select = await screen.findByRole('button', {name: /Select year/});
        await user.click(select)
        const option = await screen.findByRole('option', {name: /21-22/})
        await user.click(option)
```

Initially I tried to use [selectOptions](https://testing-library.com/docs/user-event/utility#-selectoptions-deselectoptions) with:
```
        const select = await screen.findByLabelText(/Select year/)
        await user.selectOptions(select, '21-22')
```

This fails because Instructure UI creates 2 elements that have a label that matches "Select year":

```
<input
  aria-describedby="Selectable__uuz3Vpujk3se-description"
  aria-expanded="false"
  aria-haspopup="listbox"
  autocomplete="off"
  class="qBMHb_cwos qBMHb_ycrn qBMHb_EMjX"
  id="Select__uefLVg3Pfy6O"
  placeholder="Select a year"
  readonly=""
  role="button"
  title=""
  type="text"
  value=""
/>
```

and

```
<button
  class="fOyUs_bGBk fOyUs_fKyb fOyUs_cuDs fOyUs_cBHs fOyUs_eWbJ fOyUs_fmDy fOyUs_eeJl fOyUs_cBtr fOyUs_fuTR fOyUs_cnfU fQfxa_bGBk"
  cursor="pointer"
  data-cid="BaseButton IconButton"
  style="margin: 0px; padding: 0px; border-radius: 0; border-width: 0px; width: auto; cursor: pointer;"
  type="button"
>
...[trimmed]...
```

**Comment by ke1k0** [comment]
Hi @buckett ! 

The Select and SimpleSelect components were made by the former InstUI team, so sadly we don't know the exact reason behind this. Luckily though, they have left a comment in Select's code, so I think I have a good idea what is happening:

```js
    const isEditable = typeof onInputChange !== 'undefined'

    // props to ensure screen readers treat uneditable selects as accessible
    // popup buttons rather than comboboxes.
    const overrideProps: Partial<TextInputProps> = !isEditable
      ? {
          role: 'button',
          title: inputValue,
          'aria-autocomplete': undefined
        }
      : {}
``` 

This sets the role prop on the underlying TextInput component. The `isEditable` variable is true when the component receives the `onInputChange` handler (e.g.: when Select has autocomplete behaviour). When the input is editable, the input has role "combobox" or "listbox". When there is no handler, the component works like a simple HTML `<select>`, and opens the list on click, the trigger is not editable.

SimpleSelect is based on Select, but is meant to work more like the original HTML `<select>` (as stated in the [docs](https://instructure.design/#SimpleSelect)), so it never passes the event handler to the underlying Select, and the trigger is never editable. This means that the override applies and the input renders with `role="button"`.

As for the comment, it states that they did this for better screen reader support. If you need the exact reason, I can ask someone from the A11y team, they might know more about this issue. :)

If you need to target the input, I'd suggest passing an id to the component as it is passed to the input:
```js
<SimpleSelect renderLabel="label" id="passIdHere">
  ...
</SimpleSelect>
```
renders:
```html
<input
  id="passIdHere"
  aria-haspopup="listbox"
  aria-expanded="false"
  aria-describedby="Selectable_38-description"
  role="button" 
  autocomplete="off" 
  readonly="" 
  title="Foo" 
  type="text" 
  class="css-wlkv2r-textInput" 
  value="Foo"
>
```

Let me know if this helped!

**Comment by ke1k0** [comment]
I'll try to ask around in the A11y team, hopefully someone remembers why it was solved like this back then. I'll get back to you when I know more!
